### PR TITLE
Migrate Terminal Panel to ViewPane

### LINKED ISSUE #89729: Enable views to move between containers (Part 2)
This is a continuation of the work from #85164.

## Debt

- [x] `ViewDescriptorService` tests #89686
- [x] Remove sync `openPanel` in favor of `async` #89723
- [ ] Fix issue with panel naming before opening #88637
- [x] Have all panels using the new infrastructure #90252 #89993 50bc5d71aa4f5ea728ace3daaa4574f12fcc4d9a
- [ ] Make sure views are not using panel or viewlet semantics (`IPanelService`, `IViewletService`) 
#90020
- [x] ~~Introduce `attachStyler` in `ViewPane` so that implementing views can simply attach tree or list using it.~~ No longer necessary

## Features
 
- [x] Ensure all core views can be moved without issue
  - [x] Explorer
    - [x] Open Editors a28e0a93b170ead69645a3d64f00e7d3a06af2d0
    - [x] ~File Explorer~
    - [x] Timeline d05b78490f04d66d05a856f57169f724d7679049
  - [x] ~~Source Control~~ #90461
  - [x] Search
  - [x] Run and Debug #90468
  - [x] Extensions (will not be enabled)
  - [x] References view?
  - [x] ~~Remote~~
- [ ] Invert `canMove` property to allow exclusion of extensions viewlet views
- [x] Drag and drop support between existing containers
- [x] Drag and drop support into new containers

## Future (Not targeted for February)

- [ ] Generate icons for panel views/containers to enable new containers in sidebar from panel views
- [ ] Enable multiple views in panels

### PR DESCRIPTION
**Pull request by sbatten**
This PR refs #89729

### CODE REVIEW

File: src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts [hunk 1]
```diff
@@ -289,7 +289,7 @@ export class TerminalTaskSystem implements ITaskSystem {
 			return false;
 		}
 		const activeTerminalInstance = this.terminalService.getActiveInstance();
-		const isPanelShowingTerminal = this.panelService.getActivePanel()?.getId() === TERMINAL_PANEL_ID;
+		const isPanelShowingTerminal = this.panelService.getActivePanel()?.getId() === TERMINAL_VIEW_ID;
```
Comment (Tyriar) [inline src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts]: Does this need to change to `isViewShowingTerminal` and views service instead?

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts [hunk 2]
```diff
@@ -964,7 +964,7 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 
 	private _refreshSelectionContextKey() {
 		const activePanel = this._panelService.getActivePanel();
-		const isActive = !!activePanel && activePanel.getId() === TERMINAL_PANEL_ID;
+		const isActive = !!activePanel && activePanel.getId() === TERMINAL_VIEW_ID;
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalInstance.ts]: Again should this not use views service?

File: src/vs/workbench/contrib/terminal/browser/terminal.contribution.ts [hunk 3]
```diff
@@ -362,15 +365,20 @@ registry.registerWorkbenchAction(SyncActionDescriptor.create(QuickOpenTermAction
...
+	name: nls.localize('terminal', "Terminal"),
+	ctorDescriptor: new SyncDescriptor(ViewPaneContainer, [TERMINAL_VIEW_ID, TERMINAL_VIEW_ID, { mergeViewWithContainerWhenSingleView: true, donotShowContainerTitleWhenMergedWithContainer: true }]),
+	focusCommand: { id: TERMINAL_COMMAND_ID.FOCUS }
+}, ViewContainerLocation.Panel);
+Registry.as<panel.PanelRegistry>(panel.Extensions.Panels).setDefaultPanelId(TERMINAL_VIEW_ID);
+
+Registry.as<IViewsRegistry>(ViewContainerExtensions.ViewsRegistry).registerViews([{
+	id: TERMINAL_VIEW_ID,
+	name: nls.localize('terminal', "Terminal"),
+	canToggleVisibility: false,
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminal.contribution.ts]: No idea what this is meant to drive, I can still right click to toggle the visibility in the panel?

**Comment by Tyriar** [comment]
Lowercase "hide" on right click? Isn't the terminal check item enough below also?

Screen Shot 2020-02-08 at 1 33 47 PM

**Comment by Tyriar** [comment]
I don't know how to move it to the viewlet, `"workbench.view.experimental.allowMovingToNewContainer": true` didn't do it.

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts [hunk 4]
```diff
@@ -655,34 +657,34 @@ export class TerminalService implements ITerminalService {
...
-		const panel = this._panelService.getActivePanel() as TerminalPanel;
-		if (panel && panel.getId() === TERMINAL_PANEL_ID) {
-			panel.hideFindWidget();
+		const pane = this._viewsService.getActiveViewWithId(TERMINAL_VIEW_ID) as TerminalViewPane;
+		if (pane) {
+			pane.hideFindWidget();
 			this._findWidgetVisible.reset();
-			panel.focus();
+			pane.focus();
 		}
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalService.ts]: This and findNext/findPrevious no longer verify if the terminal is active or not, I guess that's fine though.

**Review by Tyriar — APPROVED** [review]
Looks great, thanks for this!

File: src/vs/workbench/contrib/terminal/browser/terminal.contribution.ts (hunk 3 above)
Comment (sandy081) [reply src/vs/workbench/contrib/terminal/browser/terminal.contribution.ts]: This shall be set to `true` unless we do not want terminal to hidden.

**Review by sandy081 — APPROVED** [review]

**Comment by sandy081** [comment]
> I don't know how to move it to the viewlet, "workbench.view.experimental.allowMovingToNewContainer": true didn't do it.

You shall enable `canMove` on the `view` while registering.

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts (hunk 4 above)
Comment (sbatten) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts]: getActiveViewWithId only returns the requested view if it is active. so if (pane) is sufficient

**Pull request merged**